modules/nf-core/krakentools/extractkrakenreads/main.nf

```diff
@@ -0,0 +1,48 @@
+process KRAKENTOOLS_EXTRACTKRAKENREADS {
+
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/krakentools:1.2--pyh5e36f6f_0':
+        'biocontainers/krakentools:1.2--pyh5e36f6f_0'}"
+
+    input:
+    val taxid // Separated by spaces
+    tuple val(meta), path(classified_reads_assignment)
+    tuple val(meta), path(classified_reads_fastq)
+    tuple val(meta), path(report)
+
+    output:
+    tuple val(meta), path("*.{fastq,fasta}"), emit: extracted_kraken2_reads
+
+    path "versions.yml", emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def output_format = task.ext.fastq_output ? "fastq" : "fasta"
```

Comment:
@Joon-Klaps Is it okay to use `task.ext.fastq_output` directely without defining the option`--fastq-output `

Reply:
I'm afraid it doesn't align with nf-core guidelines, you can use do something similar to [bcftools](https://github.com/nf-core/modules/blob/8a2dd9e84cde68ebba45fb2c602c312c999c02a0/modules/nf-core/bcftools/annotate/main.nf#L25) 
```suggestion
    extension = args.contains("--fastq-output") ? "fastq" : "fasta"
```

Reply:
Done! Thanks for very good suggestions.